lib/rules/no-hooks-for-single-case.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var astUtil = require('../util/ast');
+
+function newDescribeLayer(describeNode) {
+    return {
+        describeNode: describeNode,
+        hookNodes: [],
+        testCount: 0
+    };
+}
+
+module.exports = function (context) {
+    var layers = [];
+
+    function popLayer(node) {
+        var layer = layers[layers.length - 1];
+        if (layer.describeNode === node) {
+            if (layer.testCount <= 1) {
+                layer.hookNodes.forEach(function (hookNode) {
+                    context.report({
+                        node: hookNode,
+                        // message: 'Unexpected use of duplicate Mocha `' + name + '` hook'
+                        message: 'Unexpected use of duplicate Mocha `before` hook'
```

Comment:
That message doesn’t seem to make sense for this rule. Shouldn’t be something like "Unnecessary use of Mocha hook for only one test case"?